### PR TITLE
feat: v1 codemods

### PR DESCRIPTION
**Pull request by jmiazga**
I am creating this PR to track what needs to be done, and hopefully others can help with this.

### General tasks
- [ ] write README
- [ ] update package.json
- [ ] determine if we can write tests against the codemods
- [ ] need to figure out how to only modify chakra components (jscodeshift findJSXElementsByModuleName method doesnt work with es modules)

### Codemods (some might not be worthwhile or feasible)
These tasks are coming from the [migration doc](https://github.com/chakra-ui/chakra-ui/blob/07bc4e864623c061070727d296d925e68a4e5d7c/docs/pages/migration.mdx)

- [ ] Update your dependencies - remove emotion from package.json (might not want to do)
- [ ] Clone Chakra's default theme (might not be worth doing, can accomplish with npx)
- [ ] Update the ThemeProvider (might not be worth doing, simple change in a single file)
- [ ] Rename variantColor to colorScheme - in progress
- [ ] Update layout size prop - in progress
- [ ] Update these style props - in progress
- [ ] ...

### CODE REVIEW

File: packages/codemods/src/variantColor-to-colorScheme.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  ["rounded", "borderRadius"],
+  ["roundedLeft", "borderLeftRadius"],
+  ["roundedRight", "borderRightRadius"],
+  ["roundedTop", "borderTopRadius"],
+  ["roundedTopLeft", "borderTopLeftRadius"],
+  ["roundedTopRight", "borderTopRightRadius"],
+  ["roundedBottomLeft", "borderBottomLeftRadius"],
+  ["roundedBottomRight", "borderBottomRightRadius"],
+  ["roundedBottom", "borderBottomRadius"],
+  ["size", "boxSize"], // do we need to change all size to boxSize or just for Box component
```
Comment (segunadebayo) [inline packages/codemods/src/variantColor-to-colorScheme.ts:17]: We need to careful with this one. Some components use the `size` prop to refer to visual size and not width+height. Just like button, checkbox, etc.

For these components, we won't change the prop. but we'll update for the rest. I guess the best solution is to keep an array of the components to ignore.

**Comment by segunadebayo** [comment]
Great work @jmiazga and @with-heart. Thank you for working on this.

For cloning the default theme, we're already creating a CLI for this. So users will only do `npx chakra init` and it'll set it up for them.

File: packages/codemods/src/variantColor-to-colorScheme.ts (hunk 1 above)
Comment (with-heart) [reply packages/codemods/src/variantColor-to-colorScheme.ts:17]: Yeah we're definitely going to use `width`+`height`. Unfortunately `jscodeshift` is kind of a pain, so we very quickly got stuck but we're going to keep working on it and see where we can get.

Comment (with-heart) [reply packages/codemods/src/variantColor-to-colorScheme.ts:17]: Oh, my comment only addressed half of your point lol. We're planning on changing the prop name only if the value of `size` isn't one of the existing t-shirt sizes we use in `master`.

**Pull request closed without merging**